nifi-nar-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/test/java/org/apache/nifi/amqp/processors/ConsumeAMQPTest.java

```diff
@@ -170,8 +180,11 @@ public void validateHeaderWithValueSeparatorForHeaderParameterConsumeAndTransfer
         final Map<String, List<String>> routingMap = Collections.singletonMap("key1", Arrays.asList("queue1", "queue2"));
         final Map<String, String> exchangeToRoutingKeymap = Collections.singletonMap("myExchange", "key1");
         final Map<String, Object> headersMap = new HashMap<>();
-        headersMap.put("foo1","bar,bar");
-        headersMap.put("foo2","bar,bar");
+        headersMap.put("foo1", "bar,bar");
+        headersMap.put("foo2", "bar,bar");
+        headersMap.put("foo3", "null");
+        headersMap.put("foo4", null);
+        String expectedResult = String.format("{%s}", ConsumeAMQP.convertMapToString(headersMap, "|"));
```

Comment:
I recommend testing against a hardcoded value because this way, we are testing that the processor called this method, but we need to know whether it is doing what we expect.